File: Technology Fundamentals/14.Exercises - Array Advanced/houseParty.js
function houseParty(input){
    let guests = [];

    for (let index = 0; index < input.length; index++) {
        const element = input[index];
        let name = element.split(' ')[0];
        if(element.includes('not going!')){
            if(guests.includes(name)){
                guests = guests.filter(el => el !== name);
            }else{
                console.log(`${name} is not in the list!`);
            }
        }else{
            if(!guests.includes(name)){
                guests.push(name);
            }else{
                console.log(`${name} is already in the list!`);
            }
        }
    }
    guests.forEach(guest => console.log(guest));
}
houseParty(['Allie is going!','George is going!',
'John is not going!', 'George is not going!'])

File: Technology Fundamentals/14.Exercises - Array Advanced/Csharp/03.HouseParty/HouseParty.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace _03.HouseParty
{
    public class HouseParty
    {
        public static void Main()
        {
            int numberCommand = int.Parse(Console.ReadLine());

            List<string> guests = new List<string>();
            for (int i = 0; i < numberCommand; i++)
            {
                string[] commandParams = Console.ReadLine()
                    .Split();

                string name = commandParams[0];
                if (commandParams.Length == 3)
                {
                    if (!guests.Contains(name))
                    {
                        guests.Add(name);
                    }
                    else
                    {
                        Console.WriteLine($"{name} is already in the list!");
                    }
                }
                else
                {
                    if (!guests.Contains(name))
                    {
                        Console.WriteLine($"{name} is not in the list!");
                    }
                    else
                    {
                        guests.Remove(name);
                    }
                }
            }

            foreach (var guest in guests)
            {
                Console.WriteLine(guest);
            }
        }
    }
}
